homeassistant/components/binary_sensor/zoneminder.py

```diff
@@ -0,0 +1,66 @@
+"""
+Support for state (idle, alarm) of ZoneMinder monitors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.zoneminder/
+"""
+import logging
+from homeassistant.components.binary_sensor import (BinarySensorDevice)
+import homeassistant.components.zoneminder as zoneminder
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['zoneminder']
+ZM_IDLE = '0'
+ZM_TAPE = '4'
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the ZoneMinder sensors."""
+    sensors = []
+
+    _LOGGER.debug("Getting monitors")
+    monitors = zoneminder.get_state('api/monitors.json')
+    for i in monitors['monitors']:
+        sensors.append(
+            ZMBinarySensor(int(i['Monitor']['Id']), i['Monitor']['Name'])
+        )
+
+    _LOGGER.debug("Adding devices")
+    add_devices(sensors)
+
+
+class ZMBinarySensor(BinarySensorDevice):
+    """Get the state of each ZoneMinder monitor."""
+
+    def __init__(self, monitor_id, monitor_name):
+        """Initiate monitor sensor."""
+        self._monitor_id = monitor_id
+        self._monitor_name = monitor_name
+        self._sensor_class = 'motion'
+        self._state = False
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} Status'.format(self._monitor_name)
+
+    @property
+    def is_on(self):
+        """Return the status of the sensor."""
+        _LOGGER.debug('is on?')
+        return self._state
+
+    def update(self):
+        """Return the current status of the monitor."""
+        _LOGGER.debug("getting status")
+        monitor = zoneminder.get_state(
+            'api/monitors/alarm/id:%i/command:status.json' % self._monitor_id
+        )
+        _LOGGER.debug("got status")
+        if monitor['status'] == ZM_IDLE:
+            self._state = False
+        elif monitor['status'] == ZM_TAPE:
+            self._state = False
+        else:
+            self._state = 'Alarm'
```

Comment:
Note that I don't have commit access so my suggestion is just that, a suggestion as another contributor and user. IMO it should be a non-binary sensor reporting all states. I personally would say to just add it to `sensor.zoneminder` for now. Users who don't want one of the existing 2 sensors or this new sensor can just hide the ones they don't care about (via `customize:`).

So I would suggest creating a new non-binary sensor class in "homeassistant/components/sensor/zoneminder.py" to add to the existing 2 and have it created in the existing loop in `setup_platform`. Make sure to handle the new API endpoint not existing yet (since 1.29 seems to be the stable version on Debian still) either by checking `/api/host/getVersion.json` or just handling the error returned by the endpoint.